protocol/messenger_handler.go

```diff
@@ -1661,11 +1661,18 @@ func (m *Messenger) HandleCommunityRequestToJoinResponse(state *ReceivedMessageS
 	}
 
 	// check if it is outdated approved request to join
-	if requestToJoin.State != communities.RequestToJoinStatePending && requestToJoinResponseProto.Clock <= community.Clock() {
+	clockSeconds := requestToJoinResponseProto.Clock / 1000
+	isClockOutdated := clockSeconds < requestToJoin.Clock
+	isDuplicateAfterMemberLeaves := clockSeconds == requestToJoin.Clock &&
+		requestToJoin.State == communities.RequestToJoinStateAccepted && !community.Joined()
+
+	if requestToJoin.State != communities.RequestToJoinStatePending &&
+		(isClockOutdated || isDuplicateAfterMemberLeaves) {
 		m.logger.Error(ErrOutdatedCommunityRequestToJoin.Error(),
 			zap.String("communityId", community.IDString()),
-			zap.Uint64("communityClock", community.Clock()),
-			zap.Uint64("requestToJoinClock", requestToJoinResponseProto.Clock),
+			zap.Bool("joined", community.Joined()),
+			zap.Uint64("requestToJoinResponseProto", requestToJoinResponseProto.Clock),
+			zap.Uint64("requestToJoin", requestToJoinResponseProto.Clock),
```

Comment:
Logging `requestToJoinResponseProto.Clock` twice.
Also, there might be some sensitive information in `requestToJoinResponseProto`, like members list, etc. Better not to log it.